backend/app/api/v1/endpoints/auth.py

```diff
@@ -0,0 +1,343 @@
+import json
+
+from fastapi import APIRouter, Depends, Form, HTTPException, Request, Response, status
+from fastapi.templating import Jinja2Templates
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core.dependency import (
+    check_authorization,
+    check_existing_email,
+    check_existing_username,
+    check_user_active,
+    check_user_permission,
+)
+from app.core.security import (
+    create_access_token,
+    generate_reset_token,
+    get_current_user,
+    get_current_user_role,
+    hash_password,
+    send_reset_email,
+    verify_old_password,
+    verify_reset_token,
+    verify_token,
+)
+from app.core.service import send_verification_email
+from app.db.crud import (
+    create_in_db,
+    fetch_data_by_id,
+    update_instance,
+    update_instance_fields,
+)
+from app.db.database import get_db
+from app.model.base_model import User
+from app.schema.auth_schema import UserInResponse
+from app.util.hash import async_hash_password, verify_password
+from logger import log
+
+router = APIRouter(prefix="/auth", tags=["Authentication:"])
+
+templates = Jinja2Templates(directory="app/templates")
+
+
+def test_create_user(test_client):
+    print("Executing test_create_user")
+    data = {
+        "username": "test_user",
+        "email": "test@example.com",
+        "password": "password123",
+        "role": "user",
+        "first_name": "Test",
+        "last_name": "User",
+        "contact_number": "1234567890",
+        "gender": "male"
+    }
+    response = test_client.post("/api/v1/auth/create-user", data=data)
+    print(f"Response status code: {response.status_code}")
+    print(f"Response content: {response.content}")
+    assert response.status_code == 201
+    response_json = response.json()
+    assert "id" in response_json
+    assert response_json["email"] == "test@example.com"
+
+
+
+@router.post("/verify")
+async def verify_email(
+    request: Request,
+    email: str = Form(...),
+    v_token: str = Form(...),
+    db: AsyncSession = Depends(get_db),
+):
+    log.info(f"Attempting to verify email: {email}")
+    verification_result = verify_token(email, v_token)
+
+    if verification_result:
+        user = await db.execute(select(User).filter(User.email == email))
+        user_instance = user.scalar_one()
+        user_instance.is_active = True
+        await db.commit()
```

Comment:
Write a base class inside the database folder. The class should include basic CRUD operations. This will allow you to reuse the db operations where required. The current approach requires a lot of ambiguous code. 

Check this code for reference. 
https://github.com/ChristopherGS/ultimate-fastapi-tutorial/blob/main/part-11-dependency-injection/app/crud/base.py

Reply:
Added a base class in db folder named crud_base.py which is inherited by the model-based crud operations written in db>crud_auth.py and db>crud_task.py . then finally called by the endpoints through HTTP requests, like this:
```
db_task = await task_crud.create(db, obj_in=task_data)
```